src/irmin-pack/inode_intf.ml

```diff
@@ -65,41 +61,31 @@ module type INTER = sig
     int ->
     int * Elt.t
 
-  module Val_impl : sig
-    type t
-
-    val pred : t -> [ `Node of hash | `Inode of hash | `Contents of hash ] list
-    val of_bin : Elt.t -> t
-    val save : add:(hash -> Elt.t -> unit) -> mem:(hash -> bool) -> t -> unit
-    val hash : t -> hash
-  end
-
   module Val : sig
     type nonrec hash = hash
-    type t = { find : hash -> Val_impl.t option; v : Val_impl.t }
+    type t
 
     include Val_intf with type hash := hash and type t := t
-  end
-end
 
-module type S_EXT = sig
-  include Irmin.CONTENT_ADDRESSABLE_STORE
-
-  val batch : [ `Read ] t -> ([ `Read | `Write ] t -> 'a Lwt.t) -> 'a Lwt.t
+    (* The following symbols are only available from codes that directly call
+       [Make_inter], e.g. [Make_ext], inode_layers.ml and test_inode.ml
+       The other pieces of code will only see [Val] as a [Val_intf], through
+       [S].
+    *)
```

Comment:
I think this is either going to be (a) obvious from the code or (b) made incorrect on a future revision. Perhaps just replace it with a doc-comment saying that this is an unstable internal API?